subprojects/model-core/model-core.gradle.kts

```diff
@@ -23,11 +23,11 @@ plugins {
 }
 
 dependencies {
+    api(project(":coreApi"))
```

Comment:
Thanks @melix for pinpointing this.

> Cédric Champeau wrote:
MapPropertyExtension belongs to model-core
it depends on MapProperty, and it's obviously an API dependency
however in model-core.gradle.kts, it's defined as an implementation dependency
so, MapProperty will not be visible to the consumers
this is exactly the error you are seeing
So I'm not sure why we don't already see the problem on master, unless the dependencies changed
I'm not quite sure what happens, but clearly the extension is loaded now and properly fails
can you try to put coreApi as an api dependency of model-core instead?
a NoClassDefFound means -> a dependency of the class I'm trying to load is missing. Which is clearly the case if you have on the compile classpath, say, class A extends B, and B is missing.
so the compiler complains "hey, dumbass, can you please give me B too?"
it says that more gently but this is what it means :slightly_smiling_face:
now I'm puzzled why your changes cause that to happen.
Either the extension wasn't in use, or the way you changed things causes a more eager initialization of the Groovy runtime.
Anyway, the declaration in model-core is wrong
it's an API dependency, because MapPropertyExtensions is a public extension of MapProperty, and therefore a public API exposes MapProperty.
feel free to open an issue, but at this stage I'd just consider this solved :slightly_smiling_face:
could be that the new version shouldn't require the extension
or could be that the old version didn't see the extension